src/parsing/binast-parse-data.cc

```diff
@@ -17,15 +17,23 @@ namespace internal {
 // A serialized BinAstParseData in zone memory (as apposed to being on-heap).
 class ZoneBinAstParseData : public ZoneObject {
  public:
-  V8_EXPORT_PRIVATE ZoneBinAstParseData(Zone* zone, Vector<uint8_t>* byte_data);
+  V8_EXPORT_PRIVATE ZoneBinAstParseData(
+      Zone* zone, Vector<uint8_t>* byte_data,
+      std::vector<BinAstInnerFunctionLiteralOffset>
+          function_literal_offsets);
 
   Handle<BinAstParseData> Serialize(Isolate* isolate);
   Handle<BinAstParseData> Serialize(OffThreadIsolate* isolate);
 
   ZoneVector<uint8_t>* byte_data() { return &byte_data_; }
 
+  std::vector<BinAstInnerFunctionLiteralOffset> function_literal_offsets() {
+    return std::move(function_literal_offsets_);
```

Comment:
re: idiomatic usage of `std::move`, I'd avoid using it on a getter like this, as it invalidates the underlying segment of data being pointed to. Prefer returning a const reference for cases like this instead.

Reply:
ah! that makes sense. thanks.